cpp/merge_print_and_online.cc

```diff
@@ -205,47 +205,55 @@ void EliminateDanglingOrUnreferencedCrossLinks(const bool debug, const std::unor
                                                std::unordered_multimap<std::string, std::string> * const canonical_ppn_to_ppn_map)
 {
     unsigned dropped_count(0);
+
+    //
+    // 1. Remove inappropriate or missing non-canonical PPN's
+    //
+
     auto canonical_ppn_and_ppn(canonical_ppn_to_ppn_map->begin());
-    std::set<std::string> group_ppns;
     while (canonical_ppn_and_ppn != canonical_ppn_to_ppn_map->end()) {
-        const std::string canonical_ppn(canonical_ppn_and_ppn->first);
-        group_ppns.emplace(canonical_ppn_and_ppn->second);
-        const auto next_canonical_ppn_and_ppn(std::next(canonical_ppn_and_ppn, 1));
-        bool drop_group(false);
-
-        if (next_canonical_ppn_and_ppn == canonical_ppn_to_ppn_map->end()
-            or next_canonical_ppn_and_ppn->first != canonical_ppn)
-        {
-            // Decide to drop group either if PPN for merging is not a superior PPN or doesn't exist...
-            drop_group = ppn_to_offset_map.find(canonical_ppn) == ppn_to_offset_map.end();
-
-            // ... or at least one of the PPN's doesn't exist
-            if (not drop_group) {
-                for (const auto &ppn : group_ppns) {
-                    if (ppn_to_offset_map.find(ppn) == ppn_to_offset_map.end()) {
-                        LOG_INFO("Don't merge group around PPN " + ppn + " because the PPN is missing in our data! All PPNs in group: "
-                                 + StringUtil::Join(group_ppns, ','));
-                        drop_group = true;
-                        break;
-                    }
-                }
-            }
+        if (ppn_to_offset_map.find(canonical_ppn_and_ppn->second) != ppn_to_offset_map.end())
+            canonical_ppn_and_ppn = std::next(canonical_ppn_and_ppn, 1);
```

Comment:
1 is default and can be omitted
https://en.cppreference.com/w/cpp/iterator/next